### DISCUSSION TITLE
CTkCheckBox command executed before changing the variable

### DISCUSSION THREAD

**Discussion by yevheniia1smortsova** [TomSchimansky/CustomTkinter#128] [General]
Dear sir,

If I understood well, the command linked to CheckBox widget is executed before the variable is set to off/onvalue (toggle method part): 

            if self.function is not None:
                self.function()

            if self.variable is not None:
                self.variable_callback_blocked = True
                self.variable.set(self.onvalue if self.check_state is True else self.offvalue)
                self.variable_callback_blocked = False

Was there a specific purpose of doing that? In my case, I wrote a command that has a variable value condition, can we change the order of these two statements?

Thanks in advance for your reply,
Yevheniia Smortsova

**Comment by TomSchimansky** [comment]
Hey, yes I can swap the order of these statements, there was no specific reason for this as far as I can remember it. It'll be changed in the next version.

**Reply by yevheniia1smortsova** [reply]
Thank you a lot!
